apps/src/templates/levelSummary/CheckForUnderstanding.jsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {ViewType} from '@cdo/apps/code-studio/viewAsRedux';
+import getScriptData from '@cdo/apps/util/getScriptData';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import SectionSelector from '@cdo/apps/code-studio/components/progress/SectionSelector';
+import i18n from '@cdo/locale';
+import styles from './check-for-understanding.module.scss';
+
+const SUMMARY_PARAM = 'view=summary';
+const FREE_RESPONSE = 'FreeResponse';
+
+const CheckForUnderstanding = ({
+  isRtl,
+  viewAs,
+  selectedSection,
+  students,
+  currentLevelId,
+  levels
+}) => {
+  const currentLevel = levels.find(l => l.activeId === currentLevelId);
+  const nextLevel = levels.find(l => l.position === currentLevel.position + 1);
+
+  // To avoid confusion, if a teacher tries to view the summary as a student,
+  // send them back to the level in Participant mode instead.
+  if (viewAs === ViewType.Participant) {
+    const paramString = document.location.search
+      .replace(SUMMARY_PARAM, '')
+      .replace('&&', '&')
+      .replace('?&', '?');
+    document.location.replace(currentLevel.url + paramString);
+  }
```

Comment:
this feels messy, but I wasn't sure a better way that handled all the cases (`url?view=summary`, `url?otherParams=blah&view=summary&moreParams=blah`, etc)